woocommerce/woocommerce
# [Product Block Editor]: move Modal editor out of the description block edit component

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This Pull Request (PR) significantly improves how our application handles editor components. It separates the Modal editor component from the Description block edit component. The primary effect of this change is that it prevents both editor instances from being rendered simultaneously in the same context. As a result, the application will now render only one editor at a time, enhancing performance and user experience.

#### Enhancements in Modal Editor Usage

Key additions in this PR include several actions integrated into the `woo/product-editor-ui` store. These actions are designed to manage the Modal editor's content. One of the actions acts like a signal that detects if the user has modified the content in the Modal editor. This functionality is essential for synchronizing content across different components of our application.
For example, it allows for seamless content integration from the Modal editor into the preview component of the Description block.

Here are the specific changes:
- The Modal editor's blocks are now stored within the product editor UI store. This centralizes the management of these blocks and simplifies their accessibility across the application.
- We have eliminated the practice of storing block instances from the iFrame editor in the local state. These instances are stored globally, aligning with our application's data management strategy.
- The Modal editor component has been relocated to be part of the regular product editor. This repositioning promotes a more cohesive user interface.

These changes aim to improve the maintainability and scalability of our editor components, ultimately enhancing the overall user experience.

Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the block editor
2. Edit the product description as usual
3. Additionally, you can use the Redux tool to see how the data is propagated through the app workflow.

https://github.com/woocommerce/woocommerce/assets/77539/35491aac-c2fa-451e-950e-a0745d23e30a




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
